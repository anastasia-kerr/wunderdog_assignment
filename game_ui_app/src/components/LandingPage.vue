<template>
  <v-container>
    <v-row class="text-center">
      <v-col class="mb-4">
        <h1 class="display-2 text-center font-weight-bold mb-3">
          {{ header }}
        </h1>
      </v-col>
    </v-row>
    <v-row class="text-center">
      <v-col class="mb-4">
        <h1 class="red--text" v-text="subtitle" />
      </v-col>
    </v-row>
    <v-row class="text-center">
      <v-col class="mb-4">
        <v-btn id="start_game_button" color="primary" to="/create-game" v-text="startLabel" />
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'LandingPage',

  data: () => ({
    header: "Welcome to Anastasiia Kerr's assignment for Wunderdog",
    subtitle: 'ROCK PAPER SCISSORS',
    startLabel: 'Start new game'
  })
})
</script>
