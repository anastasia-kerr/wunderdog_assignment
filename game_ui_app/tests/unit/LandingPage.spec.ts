import Vue from 'vue'
import LandingPage from '@/components/LandingPage.vue'
import { mount, createLocalVue, Wrapper } from '@vue/test-utils'
import Vuetify from 'vuetify'

const localVue = createLocalVue()

describe('LandingPage', () => {
  let wrapper:Wrapper<Vue>
  beforeEach(() => {
    const vuetify = new Vuetify()

    wrapper = mount(LandingPage, { localVue, vuetify })
  })
  it('Must have a button to create a game', () => {
    const btn = wrapper.find('#start_game_button')

    expect(btn.exists()).toBe(true)
    expect(btn.text()).toContain('Start new game')
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
