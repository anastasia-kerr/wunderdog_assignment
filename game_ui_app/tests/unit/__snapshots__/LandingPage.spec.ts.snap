// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
<v-container>
  <v-row
    class="text-center"
  >
    <v-col
      class="mb-4"
    >
      <h1
        class="display-2 text-center font-weight-bold mb-3"
      >
        
        Welcome to Anastasiia Kerr's assignment for Wunderdog
      
      </h1>
    </v-col>
  </v-row>
   
  <v-row
    class="text-center"
  >
    <v-col
      class="mb-4"
    >
      <h1
        class="red--text"
      >
        ROCK PAPER SCISSORS
      </h1>
    </v-col>
  </v-row>
   
  <v-row
    class="text-center"
  >
    <v-col
      class="mb-4"
    >
      <v-btn
        color="primary"
        id="start_game_button"
        to="/create-game"
      >
        Start new game
      </v-btn>
    </v-col>
  </v-row>
</v-container>
`;
